staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -129,9 +130,31 @@ func ValidateNoNewFinalizers(newFinalizers []string, oldFinalizers []string, fld
 
 // ValidateImmutableField validates the new value and the old value are deeply equal.
 func ValidateImmutableField(newVal, oldVal interface{}, fldPath *field.Path) field.ErrorList {
+	return ValidateImmutableFieldWithErr(newVal, oldVal, field.Invalid(fldPath, newVal, FieldImmutableErrorMsg))
+}
+
+// ValidateImmutableFieldWithErr validates the new value and the old value are deeply equal.
+// It accepts an err *field.Error parameter as base error. If the result is not equal, ValidateImmutableFieldWithErr will append additional error information
+// in err.detail and return this error.
+func ValidateImmutableFieldWithErr(newVal, oldVal interface{}, err *field.Error) field.ErrorList {
 	allErrs := field.ErrorList{}
+	defer func() {
+		if r := recover(); r != nil {
+			// If we ever get here, then either DeepEqual or cmp.Diff failed.
+			// The former is unlikely and indicates a serious problem with the data,
+			// the latter only gets called when we already know that there is a difference.
+			// In both cases we reject the new value.
+			allErrs = append(allErrs, err)
```

Comment:
This should use neither the pre-fabricated error (current API) nor the type and message prefix (revised API), but rather generate an [InternalError](https://github.com/kubernetes/apimachinery/blob/v0.26.0/pkg/util/validation/field/errors.go#L270) [¶](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation/field#InternalError).

Reply:
Fixed